int32 state
int32 score1
int32 score2
int32 p1_training
int32 p2_training